### PR TITLE
[OFF-1281] Improve Hatch client to avoid having to duplicate the Swagger documentation

### PR DESCRIPTION
**Pull request by MichalLytek**
https://birdie.atlassian.net/browse/OFF-1281

As described in refinement:
https://www.notion.so/birdie/Uses-Hatch-generated-clients-types-in-Swagger-documentation-aka-a-much-easier-gateway-e4ce2c5656c549b0a90334367ff8ced0

Here we're getting all the files following the pattern `*.public-models.ts` and putting them in the same folder `/clients/models`.

So when we bundle the package, we should be able to do this:
```ts
import { IdentifierMapping } from "@birdiecare/rostering-integrations-client/models/identifier-mapping.public-models.ts";
```

It's not exported from main `@birdiecare/rostering-integrations-client` index file because that would collide with the usage of the client in the browser that do not have nest swagger package.

### CODE REVIEW

File: generate-client.sh [hunk 1]
```diff
@@ -30,9 +30,26 @@ then
 fi
 echo "[SUCCESS]"
 
+echo "Cloning source code repository from GitHub..."
+git clone --depth 1 --single-branch --branch staging $REPO_URL /repo
```
Comment (MichalLytek) [inline generate-client.sh:35]: @hgrsd How can I provide the `REPO_URL` like we do for `$INPUT_NAME` via `args` or `action.yml`? 🤔

**Review by Tketa — COMMENTED** [review]
Can you explain a bit in the description? :D 

Here you're getting all the files following the pattern `*.models.ts` and putting them in the same folder `/clients/models` right?

What are the next steps?

**Comment by MichalLytek** [comment]
The next step is to find a way to provide the repo url (manually or via github action magic).
Then we can merge this, expose some model in rostering integrations (identifier mapping), then use the new hatch version and check how it works.

I've tested that part of script locally and it works ok. We then need to test the whole pipeline and integration. I have some errors while trying to create the package from hatch script locally.

File: generate-client.sh (hunk 1 above)
Comment (MichalLytek) [reply generate-client.sh:35]: @djbeaumont Or maybe you can help me 😉

Comment (djbeaumont) [reply generate-client.sh:35]: So to do this you can add to the args array in `action.yaml`. However, surely there's a better way of doing this where the repo is already available to the action without needing to manually pull it. 🤔

Comment (djbeaumont) [reply generate-client.sh:35]: If the question is where does it come from - it does come from action.yaml, but the convention is that `name` in action.yaml becomes available inside the action as `INPUT_NAME`. (i.e. input to the action)

File: generate-client.sh [hunk 2]
```diff
@@ -30,9 +30,26 @@ then
...
 
-echo "Creating package.json from template..."
+echo "Searching for models files..."
+cd /repo/src
+globs=$(find . -iname '*.models.ts')
```
Comment (djbeaumont) [inline generate-client.sh]:
```suggestion
globs=$(find . -iname '*.public-models.ts')
```
Maybe? Just to make it extra explicit?

File: generate-client.sh (hunk 1 above)
Comment (MichalLytek) [reply generate-client.sh:35]: Right, but how the name translates to the action yaml in target repo?
```
        with:
          path: https://rostering-integrations.staging.birdie.care/docs.json
          name: rostering-integrations
          repo_url: git@github.com:birdiecare/rostering-integrations.git
```
Like `repo_url` gonna be converted to `INPUT_REPO_URL` in shell script automatically?

Comment (djbeaumont) [reply generate-client.sh:35]: Yes I believe so 👍

Comment (djbeaumont) [reply generate-client.sh:35]: Is it possible to do this without requiring a new configuration parameter - looks like github makes the repo available as another env var already: https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables

Comment (djbeaumont) [reply generate-client.sh:35]: @MichalLytek I think I commented as you were marking the thread as resolved ☝️  looks like we can get the repo URL (or at least the path part) using an existing environment variable, so we don't have to require users of hatch to provide it.

Comment (MichalLytek) [reply generate-client.sh:35]: > However, surely there's a better way of doing this where the repo is already available to the action without needing to manually pull it.

I know but I don't have time for such research and the optimization can be done later 😉

Comment (MichalLytek) [reply generate-client.sh:35]: So we should be able to use `REPO_URL=git@github.com:${GITHUB_REPOSITORY}.git`?

Comment (MichalLytek) [reply generate-client.sh:35]: I wonder if the github action has permission to clone the repo 😄 
I think yes because we use the checkout action normally

Comment (djbeaumont) [reply generate-client.sh:35]: I hope so 😬 hopefully we do, as we have a token that has permission to push the packages back to the repo.

**Review by djbeaumont — APPROVED** [review]
Looks legit, let's give it a try!

**Comment by djbeaumont** [comment]
I wonder if we'll need to compile the models as otherwise we'll have to amend the monolith webpack configuration to transpile every client package we use models from?
Edit: hopefully the existing build takes care of that.

**Comment by MichalLytek** [comment]
@djbeaumont `tsc` is enough to compile the models - they should only have nest swagger and class-validator imports, so as no 3rd party files imported, it should just work 😉

**Review by Tketa — APPROVED** [review]

**Pull request merged**